Building pdns, dnsdist on mips(el) needs -latomic

Debian build log here: https://buildd.debian.org/status/fetch.php?pkg=pdns&arch=mipsel&ver=4.0.0%7Ealpha2-1&stamp=1456685999

```
libtool: link: mipsel-linux-gnu-g++ -DSYSCONFDIR=\"/etc/powerdns\" -DPKGLIBDIR=\"/usr/lib/mipsel-linux-gnu/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -fstack-protector-strong -Wformat -Werror=format-security "-DPACKAGEVERSION=\"4.0.0~alpha2-1.Debian\"" -std=gnu++11 -pie -pthread -Wl,-z -Wl,relro -Wl,-z -Wl,now -Wl,-z -Wl,relro -o notify arguments.o base32.o base64.o dns.o dnslabeltext.o dnsname.o dnsparser.o dnsrecords.o dnswriter.o logger.o misc.o notify.o nsecrecords.o qtype.o rcpgenerator.o selectmplexer.o sillyrecords.o statbag.o unix_utility.o  -lcrypto -lboost_program_options -pthread
dumresp.o: In function `std::__atomic_base<unsigned long long>::load(std::memory_order) const':
/usr/include/c++/5/bits/atomic_base.h:396: undefined reference to `__atomic_load_8'
/usr/include/c++/5/bits/atomic_base.h:396: undefined reference to `__atomic_load_8'
/usr/include/c++/5/bits/atomic_base.h:396: undefined reference to `__atomic_load_8'
/usr/include/c++/5/bits/atomic_base.h:396: undefined reference to `__atomic_load_8'
/usr/include/c++/5/bits/atomic_base.h:396: undefined reference to `__atomic_load_8'
dumresp.o:/usr/include/c++/5/bits/atomic_base.h:396: more undefined references to `__atomic_load_8' follow
dumresp.o: In function `std::__atomic_base<unsigned long long>::fetch_add(unsigned long long, std::memory_order)':
/usr/include/c++/5/bits/atomic_base.h:514: undefined reference to `__atomic_fetch_add_8'
/usr/include/c++/5/bits/atomic_base.h:514: undefined reference to `__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
Makefile:2090: recipe for target 'dumresp' failed
```

Adding -latomic to LDFLAGS fixed this.

Initially this issue was only about pdns, but dnsdist is also affected. No idea about recursor yet.

--- a/m4/pdns_check_os.m4
+++ b/m4/pdns_check_os.m4
@@ -35,6 +35,12 @@ AC_DEFUN([PDNS_CHECK_OS],[
   AM_CONDITIONAL([HAVE_LINUX], [test "x$have_linux" = "xyes"])
   AM_CONDITIONAL([HAVE_SOLARIS], [test "x$have_solaris" = "xyes"])
 
+  case "$host" in
+  mips*)
+    LDFLAGS="-latomic $LDFLAGS"
+    ;;
+  esac
+
   AC_SUBST(THREADFLAGS)
   AC_SUBST([DYNLINKFLAGS], [-export-dynamic])
 ])

